backend/pkg/kafka/config_helper.go

```diff
@@ -92,9 +93,18 @@ func NewKgoConfig(cfg *config.Kafka, logger *zap.Logger, hooks kgo.Hook) ([]kgo.
 
 		// OAuth Bearer
 		if cfg.SASL.Mechanism == config.SASLMechanismOAuthBearer {
-			mechanism := oauth.Auth{
-				Token: cfg.SASL.OAUth.Token,
-			}.AsMechanism()
+			var mechanism sasl.Mechanism
+			if cfg.SASL.OAUth.TokenEndpoint != "" {
```

Comment:
Alternatively we could add another config parameter called `TokenType` with options `ShortLived` and `LongLived` but I personally like this better.

Reply:
Yes seems reasonable 👍